### PR TITLE
Move to GitHub actions

### PR DESCRIPTION
**Pull request by cpcloud**
This PR moves `avro-rs` to GitHub actions.

Here's an example of what a CI run looks like, using an earlier version of this
branch against my fork of `avro-rs`: https://github.com/cpcloud/avro-rs/pull/1/checks.

GitHub actions is IMO far easier to use, more flexible with many useful
plugins, and you don't have to leave GitHub to use it.

I've added 4 builds (stable, beta, night, and an MSRV somewhat arbitrarily set
at 1.40) and 1 security audit workflow.

I'm happy to do the security audit job in another PR if that's preferable.

TODO:

- [x] Update `README.md` with correct Build Status badge

### CODE REVIEW

**Comment by cpcloud** [comment]
@poros I think you'll have to enable GitHub actions for the repo for this PR to show any checks from actions.

**Comment by poros** [comment]
Ah, I have never used them (we use Jenkins and Travis at work), but this could be a good learning opportunity. Sorry if this PR will take a bit more to come through, I have to budget some time to read about GitHub actions.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+          - 1.40.0  # MSRV
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
```
Comment (poros) [inline .github/workflows/ci.yml:22]: I see that this action latest version is 2 (https://github.com/actions/checkout). Is there any specific reason why is it better to stick with v1?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: -- -D warnings
+
+      - name: Test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test --all-features --all-targets
```
Comment (poros) [inline .github/workflows/ci.yml:52]: Do you think we could add `bench --no-run` as an additional step to make sure that the benchmarks still compile OK after a change?

File: .github/workflows/audit.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Security audit
```
Comment (poros) [inline .github/workflows/audit.yml:1]: The README of this repo suggests about specifying `paths` to speed things up (https://github.com/actions-rs/audit-check#audit-changes). Is this something you think would matter here?

**Comment by cpcloud** [comment]
> > @poros I think you'll have to enable GitHub actions for the repo for this PR to show any checks from actions.
> 
> I honestly cannot figure out how to do this. I am not the owner of the repo, so I might not have permissions? Although I seem to be able to configure a workflow for this repo from the `Actions` tab...

I think this requires permissions from @flavray. @flavray Can you give @poros full owner rights over the repo?

**Comment by poros** [comment]
It seems that ownership can only be transferred, not shared: https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/permission-levels-for-a-user-account-repository

Anyway, github actions are not listed in the above page. Perhaps we first need to merge to master to setup the workflow before we can see any action?

**Comment by poros** [comment]
Whenever you feel this PR is ready to merge @cpcloud , we could try and see how it goes.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (cpcloud) [reply .github/workflows/ci.yml]: Oh, wait nevermind. It won't.

Comment (cpcloud) [reply .github/workflows/ci.yml]: Fixed.

**Comment by cpcloud** [comment]
@poros Let's merge this!

**Comment by poros** [comment]
Alright, let's try it out. If it doesn't work, I'll revert.

**Pull request merged**